### PR TITLE
Fix `DamageContext##findMatchingCheckContext`

### PR DESCRIPTION
**Pull request by In3luki**
This was caught by a Biome linter test run.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
So it seems the condition before was only same slug and kinda same usage (is melee). Is there a chance that it might to remove the uuid check. I'm wondering if there's a scenario where this fix might end up breaking something like RE strikes.

**Comment by In3luki** [comment]
oh right, removing the check would probably be better as it didn't produce any noticeable bugs. so don't fix what's not broken.

**Comment by stwlam** [comment]
Instead of removing the broken check, why not fix it?

File: src/module/actor/roll-context/damage.ts [hunk 1]
```diff
@@ -47,11 +47,7 @@ class DamageContext<
                 const paramsItemSlug = item.slug ?? sluggify(item.name);
                 const messageItemSlug = messageItem.slug ?? sluggify(messageItem.name);
 
-                return !!(
-                    paramsItemSlug === messageItemSlug &&
-                    item.uuid === item.uuid &&
-                    item.isMelee === messageItem.isMelee
-                );
+                return !!(paramsItemSlug === messageItemSlug && item.isMelee === messageItem.isMelee);
```
Comment (stwlam) [inline src/module/actor/roll-context/damage.ts:50]:
```suggestion
                return !!(
                    paramsItemSlug === messageItemSlug &&
                    item.uuid === messageItem.uuid &&
                    item.isMelee === messageItem.isMelee
                );
```

**Review by stwlam — CHANGES_REQUESTED** [review]

**Comment by CarlosFdez** [comment]
More specifically I was hinting that maybe the uuid comparison should be tested with things like strikes from rule elements. Perhaps with strikes from battlezoo dragons that do some weird override thing.

File: src/module/actor/roll-context/damage.ts [hunk 2]
```diff
@@ -16,8 +16,7 @@ class DamageContext<
     constructor(params: DamageContextConstructorParams<TSelf, TStatistic, TItem>) {
         super(params);
         // In case the user rolled damage from their sheet, try to fish out the check context from chat
-        params.checkContext ??= this.#findMatchingCheckContext();
-
+        params.checkContext ??= this.#findMatchingCheckContext(params.options);
         if (params.outcome) params.options.add(`check:outcome:${sluggify(params.outcome)}`);
```
Comment (In3luki) [inline src/module/actor/roll-context/damage.ts:20]: I'm wondering if this was supposed to be the value of `checkContext.outcome`.

**Comment by In3luki** [comment]
Tested and works with:
- Melee Weapon
- Unarmed Attack
- Ranged Weapon
- Elemental Blast

File: src/module/actor/roll-context/damage.ts (hunk 2 above)
Comment (stwlam) [reply src/module/actor/roll-context/damage.ts:20]: Likely not

File: src/module/actor/roll-context/damage.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ class DamageContext<
 
                 return !!(
```
Comment (stwlam) [inline src/module/actor/roll-context/damage.ts:50]:
```suggestion
                return (
```

**Review by stwlam — APPROVED** [review]

**Pull request merged**